### PR TITLE
Fix bug #1435 (data loss when dumping payloads to JSON) v3

### PR DESCRIPTION
**Pull request by gozzy**
Fixed #1455 - use MemBufferWriteRaw() wrapper instead of memcpy. Also fixed MemBufferWriteRaw() - there was unneeded return stetement.

### CODE REVIEW

File: src/output-json-alert.c [hunk 1]
```diff
@@ -287,27 +285,30 @@ static int AlertJson(ThreadVars *tv, JsonAlertLogThread *aft, const Packet *p)
...
-                PrintStringsToBuffer(packet_buf, &offset,
-                                     p->payload_len + 1,
-                                     p->payload, p->payload_len);
-
                 if (json_output_ctx->flags & LOG_JSON_PAYLOAD_BASE64) {
-                    unsigned long len = sizeof(packet_buf) * 2;
+                    size_t len = p->payload_len * 2 + 1;
                     unsigned char encoded[len];
-                    Base64Encode(packet_buf, offset, encoded, &len);
+                    Base64Encode(p->payload, p->payload_len, encoded, &len);
```
Comment (inliniac) [inline src/output-json-alert.c]: output-json-alert.c: In function ‘AlertJson’:
output-json-alert.c:301:21: error: passing argument 4 of ‘Base64Encode’ from incompatible pointer type [-Werror]
util-crypt.h:82:5: note: expected ‘long unsigned int *’ but argument is of type ‘size_t *’
cc1: all warnings being treated as errors

Comment (gozzy) [inline src/output-json-alert.c]: It's a bit strange - my compiler doesn't see anything suspicious.
Maybe we should return unsigned long here? For instance, LogTlsLogPem() uses unsigned long when calling Base64Encode.

Comment (inliniac) [inline src/output-json-alert.c]: Probably yeah. size_t is 4 bytes for 32bit args, 8bytes for 64bit args.

**Comment by gozzy** [comment]
Replaced with #1468

**Pull request closed without merging**